js/private/js_run_devserver.bzl

```diff
@@ -21,14 +21,33 @@ _attrs = dicts.add(js_binary_lib.attrs, {
     "command": attr.string(),
 })
 
+def _file_to_entry_json(f):
+    short_path = f.short_path
+
+    if "/.aspect_rules_js/" in short_path:
+        # Special handling for package store deps; we only include 1st party deps since copying
+        # all 3rd party node_modules over is expensive for typical graphs
+        path_segments = f.path.split("/")
+        package_name_segment = path_segments.index(".aspect_rules_js") + 1
+
+        # TODO: @0.0.0 is by default the version of all 1p linked packages, however, it can be overridden by users
+        # if they are manually linking a 1p package and not using workspace. A more robust solution would be to
+        # split handling of 1p and 3p package in the JsInfo provider itself. Other optimizations in the rule set
+        # could also be made if that was the case.
+        if not (len(path_segments) > package_name_segment and "@0.0.0" in path_segments[package_name_segment]):
+            return None
+
+    return json.encode(short_path)
```

Comment:
Curious if this is faster than '"%s"' % short_path or the simplest concat?

Reply:
I'm not sure, I was copying `js_image_layer`...